src/equations/shallow_water_1d.jl

```diff
@@ -349,6 +426,22 @@ end
 end
 
 
+# Specialized `FluxHydrostaticReconstruction` to compute the conservative part of the numerical surface flux
+# and guarantee the approximation is well-balanced.
+# !!! warning "Experimental code"
+#     This numerical flux is experimental and may change in any future release.
+@inline function (numflux::FluxHydrostaticReconstruction)(u_ll, u_rr, orientation_or_normal_direction,
+                                                          equations::ShallowWaterEquations1D)
+  @unpack numerical_flux, hydrostatic_reconstruction = numflux
+
+  # Create the reconstructed left/right solution states in conservative form
+  u_ll_star, u_rr_star = hydrostatic_reconstruction(u_ll, u_rr, equations)
+
+  # Use the reconstructed states to compute the numerical surface flux
+  return numerical_flux(u_ll_star, u_rr_star, orientation_or_normal_direction, equations)
+end
+
+
```

Comment:
I think this looks quite general and should go into the `numerical_fluxes.jl` (without specialization on the `equations`). Or do I miss something?

Reply:
I went back and forth on where to put it. I wasn't sure what specialization meant in this context. This is specific to SWE, but is dimension agnostic. In `numerical_fluxes.jl` would I use `equations::AbstractEquations`?

Reply:
I would just write `equations` without any type restriction as in https://github.com/trixi-framework/Trixi.jl/blob/25a5cd76123fefd2383206d59b1d5f8c0ed10fb9/src/equations/numerical_fluxes.jl#L38